lib/util/ssr.js

```diff
@@ -6,13 +6,45 @@
  */
 'use strict';
 
+let ssrEscapeFound = false;
+
 module.exports.isSSREscape = function isSSREscape(node) {
-    return (
-        (node.type === 'IfStatement' || node.type === 'ConditionalExpression') &&
-        (isMetaEnvCheck(node.test) || isWindowOrDocumentCheck(node.test))
-    );
+    if (node.type === 'IfStatement' || node.type === 'ConditionalExpression') {
+        ssrEscapeFound = false;
+        if (checkConditionalStatements(node.test) || isWindowOrDocumentCheck(node.test)) {
+            return true;
+        }
+    }
+
+    return false;
 };
 
+function checkConditionalStatements(test) {
+    let node = test;
+
+    // Recursive Case: If the node is a logical expression, check its left and right parts.
+    if (node.type === 'LogicalExpression' || node.type === 'BinaryExpression') {
```

Comment:
For the LogicalExpression, should we check for the operator to be `&&` since `||` is not really correct here?

As for `BinaryExpression` – what are we looking for here? I assume we're not interested in bitwise operators like `>` and `>>`. 🙂